spec/requests/api/v1/visitor_can_pay_for_subscription_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'stripe_mock'
+
+RSpec.describe Api::V1::SubscriptionsController, type: :request do
+  let(:stripe_helper) { StripeMock.create_test_helper }
+  before(:each) { StripeMock.start }
+  after(:each) { StripeMock.stop }
+
+  let(:new_subscriber) { create(:user) }
+  let(:credentials) { new_subscriber.create_new_auth_token }
+  let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(credentials) }
+
+  describe 'Payment posts successfully' do
+    before do
+      post '/api/v1/subscriptions', 
+      params: { stripeToken: StripeMock.generate_card_token },
+      headers: headers
+    end
+
+    it 'return status 200' do
+      expect(response.status).to eq 200
+    end
+
+    it 'returns success message' do
+      expect(response_json['message']).to eq 'Your payment was successful'
+    end
+
+    it 'user role changes to subscriber when payment is successful' do 
+      allow(new_subscriber).to receive(:role).and_return('subscriber')
```

Comment:
can you explain this line?

Reply:
We took from the ATM challenge this line of code so we could update the new_subscriber Model

Reply:
It's not really relevant here

Reply:
you are not really testing the request and its response here. Feels more like a unit spec

Reply:
But we want to make sure that when a payment is successful the user roles is changed to subscriber. That's part of the feature and thus within scope.

Reply:
Well, then pull out the user from the db and assert that he is a subscriber.